File: 928-surface-area-of-3d-shapes/surface-area-of-3d-shapes.java
class Solution {
    public int surfaceArea(int[][] grid) {

        int n = grid.length;
        int area = 0;

        for (int i = 0; i < n; i++) {

            for (int j = 0; j < n; j++) {

                int value = grid[i][j];

                if (value > 0) {

                    area += value * 6;

                    area -= 2 * (value - 1);

                    if (i > 0) {
                        area -= 2 * Math.min(
                            value,
                            grid[i - 1][j]
                        );
                    }

                    if (j > 0) {
                        area -= 2 * Math.min(
                            value,
                            grid[i][j - 1]
                        );
                    }
                }
            }
        }

        return area;
    }
}

File: 928-surface-area-of-3d-shapes/surface-area-of-3d-shapes.py
class Solution:
    def surfaceArea(self, grid):
        n = len(grid)
        area = 0

        for i in range(n):
            for j in range(n):

                value = grid[i][j]

                if value > 0:
                    area += value * 6

                    area -= 2 * (value - 1)

                    if i > 0:
                        area -= 2 * min(
                            value,
                            grid[i - 1][j]
                        )

                    if j > 0:
                        area -= 2 * min(
                            value,
                            grid[i][j - 1]
                        )

        return area